### ISSUE TITLE
Prepare Investment Pools for Timelock support

### ISSUE THREAD

**Issue by nventuro** [balancer-labs/balancer-v2-monorepo#675]
Some permissioned actions are quite powerful, and it'd be best to have _some_ security mechanisms in place in case of e.g. compromised admin keys. A robust way to achieve this would be to put these critical actions behind a timelock contract.

I propose we rely on [OpenZeppelin's `TimelockController`](https://docs.openzeppelin.com/contracts/3.x/api/access#TimelockController) for this: a very simple standalone contract that forwards calls it receives, but only after a minimum amount of time has passed. To properly support this, we could split admin actions into two (or more) sets: those that are low-risk (e.g. collecting management fees - https://github.com/balancer-labs/balancer-v2-monorepo/issues/676), and those that are critical (removing tokens from the Pool - https://github.com/balancer-labs/balancer-v2-monorepo/issues/677), and setting different roles for each (e.g. supervisor and admin roles). The admin would then be a timelock: we could even do this ourselves in contract creation.

**Comment by nventuro** [comment]
Closing in favor of the approach described in https://github.com/balancer-labs/balancer-v2-monorepo/issues/876: timelocks would be implemented at the owner contract level.
